### PR TITLE
Issue #3916 - Fix whitespace between boundary and part headers

### LINKED ISSUE #3916: multipart/byterange output is invalid to RFC7233
The output of our `ResourceService` for `multipart/byterange` responses is invalid.
Discovered while testing PR #3910  for Issue #3840 

Example from Jetty 9.2.28.v20190418 server with WebAppContext and static content ...

**Request**
```
GET http://localhost:8080/traditional/small.dat HTTP/1.1
Host: localhost:8080
Connection: close
Accept: */*
Accept-Encoding: gzip
User-Agent: Jetty/9.4.19.v20190610
Range: bytes=2275-4550,6825-9100,11375-13650,15925-18200
```

**Response**
```
HTTP/1.1 206 Partial Content
Date: Wed, 31 Jul 2019 00:05:00 GMT
Last-Modified: Mon, 29 Jul 2019 19:53:41 GMT
Accept-Ranges: bytes
Content-Type: multipart/byteranges; boundary=jetty670757712jyqhnrki
Content-Length: 9416
Connection: close
Server: Jetty(9.2.28.v20190418)

--jetty670757712jyqhnrki

Content-Range: bytes 2275-4550/20480

xxxxx(...snip...)xxxxx
--jetty670757712jyqhnrki

Content-Range: bytes 6825-9100/20480

xxxxx(...snip...)xxxxx
--jetty670757712jyqhnrki

Content-Range: bytes 11375-13650/20480

xxxxx(...snip...)xxxxx
--jetty670757712jyqhnrki

Content-Range: bytes 15925-18200/20480

xxxxx(...snip...)xxxxx
--jetty670757712jyqhnrki--
```

According to https://tools.ietf.org/html/rfc7233 there should be no extra lines after the boundary and before the part headers.

### PR DESCRIPTION
**Pull request by joakime**
Signed-off-by: Joakim Erdfelt <joakim.erdfelt@gmail.com>

### CODE REVIEW

File: jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java [hunk 1]
```diff
@@ -104,15 +104,19 @@ public void startPart(String contentType)
...
+        out.write(DASHDASH);
         out.write(boundaryBytes);
-        out.write(__CRLF);
+        out.write(CRLF);
         if (contentType != null)
+        {
             out.write(("Content-Type: " + contentType).getBytes(StandardCharsets.ISO_8859_1));
-        out.write(__CRLF);
-        out.write(__CRLF);
+            out.write(CRLF);
```
Comment (joakime) [inline jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java:117]: This is the heart of the issue. (an extra CRLF)

File: jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java [hunk 2]
```diff
@@ -126,20 +130,22 @@ public void startPart(String contentType, String[] headers)
...
+        {
             out.write(("Content-Type: " + contentType).getBytes(StandardCharsets.ISO_8859_1));
-        out.write(__CRLF);
+            out.write(CRLF);
+        }
         for (int i = 0; headers != null && i < headers.length; i++)
         {
             out.write(headers[i].getBytes(StandardCharsets.ISO_8859_1));
-            out.write(__CRLF);
+            out.write(CRLF);
```
Comment (joakime) [inline jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java:146]: This is the heart of the issue. (an extra CRLF)

File: jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java [hunk 3]
```diff
@@ -126,20 +130,22 @@ public void startPart(String contentType, String[] headers)
...
-        out.write(__DASHDASH);
+        out.write(DASHDASH);
         out.write(boundaryBytes);
-        out.write(__CRLF);
+        out.write(CRLF);
         if (contentType != null)
+        {
             out.write(("Content-Type: " + contentType).getBytes(StandardCharsets.ISO_8859_1));
-        out.write(__CRLF);
+            out.write(CRLF);
```
Comment (joakime) [inline jetty-util/src/main/java/org/eclipse/jetty/util/MultiPartOutputStream.java:141]: This is the heart of the issue. (an extra CRLF)

**Comment by joakime** [comment]
I need to add some test cases for this.

**Review by gregw — APPROVED** [review]

**Pull request merged**